test/fixtures/branched/web.yml.erb

```diff
@@ -0,0 +1,32 @@
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: <%= branch %>-web
+  labels:
+    app: branched
+    branch: <%= branch %>
+  annotations:
+    shipit.shopify.io/restart: "true"
+spec:
+  replicas: 1
+  progressDeadlineSeconds: 20
+  selector:
+    matchLabels:
+      name: web
```

Comment:
I believe it would not, because that selector applies to the top-level resource, the Deployment in this case which doesn't have that label

Reply:
Things only get pruned when they're deployed by apply or create right? I think kubectl apply looks at an annotation. Because otherwise it would always prune everything if you don't pass in a selector